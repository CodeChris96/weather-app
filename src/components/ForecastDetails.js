import React from "react";

function ForecastDetails({ forecast }) {
  return (
    <div className="forecast-details">
      <p>Humidity {forecast.humidity}</p>
      <p>Wind Speed {forecast.wind.speed}</p>
      <p>Wind Direction {forecast.wind.direction}</p>
      <p>Temp Min {forecast.temperature.min}</p>
      <p>Temp Max {forecast.temperature.max}</p>
    </div>
  );
}
export default ForecastDetails;
