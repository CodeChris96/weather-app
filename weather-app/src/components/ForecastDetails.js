import React from "react";
import ForecastSummary from "./ForecastSummary";

function ForecastDetails({ forecasts = [] }) {
  return (
    <div className="forecast-details">
      {forecasts.map((forecast) => (
        <ForecastSummary
          key={forecast.date}
          date={forecast.date}
          temperatureMax={forecast.temperature.max}
          temperatureMin={forecast.temperature.min}
          humidity={forecast.humidity}
          windSpeed={forecast.wind.speed}
          windDirection={forecast.wind.direction}
        />
      ))}
    </div>
  );
}
export default ForecastDetails;
