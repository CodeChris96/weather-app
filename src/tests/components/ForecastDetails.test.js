import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    humidity: 71,
    temperature: {
      min: 12,
      max: 22,
    },
    wind: {
      speed: 13,
      direction: "s",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecasts={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("displays the correct forecast details", () => {
    const { getByText } = render(<ForecastDetails forecasts={validProps} />);
    // expect(getByText(`Humidity: ${validProps.humidity}`)).toBeInTheDocument();
    expect(
      getByText(`Wind Speed ${validProps.wind.speed}`)
    ).toBeInTheDocument();
    expect(
      getByText(`Wind Direction ${validProps.wind.direction}`)
    ).toBeInTheDocument();
    expect(
      getByText(`Temp Min ${validProps.temperature.min}`)
    ).toBeInTheDocument();
    expect(
      getByText(`Temp Max ${validProps.temperature.max}`)
    ).toBeInTheDocument();
  });
});
