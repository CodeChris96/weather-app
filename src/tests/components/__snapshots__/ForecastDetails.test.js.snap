// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    console.log(forecasts)
    <p>
      Humidity 71
    </p>
    <p>
      Wind Speed 13
    </p>
    <p>
      Wind Direction s
    </p>
    <p>
      Temp Min 12
    </p>
    <p>
      Temp Max 22
    </p>
  </div>
</DocumentFragment>
`;
